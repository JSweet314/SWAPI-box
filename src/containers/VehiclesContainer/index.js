import React, {Component} from 'react';
import fetchCategoryData from '../../apiCalls/fetchCategoryData';
import loadingGIF from '../../images/Loading_icon.gif';
import PropTypes from 'prop-types';
import VehicleCard from '../../components/VehicleCard/index';
import './style.css';

export default class VehiclesContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      vehiclesArray: [],
      pageNumber: 1,
      loading: true
    };
    this.favorites = props.favorites;
  }

  componentDidMount = () => {
    const priorData = localStorage.getItem('SWAPI-Vehicles');
    if (priorData) {
      const vehiclesArray = JSON.parse(priorData);
      this.setState({vehiclesArray, loading: false});
    } else {
      this.getVehiclesData();
    }
  }

  componentDidUpdate = (prevProps, prevState) => {
    const {vehiclesArray} = this.state;
    if (prevState.vehiclesArray !== vehiclesArray) {
      localStorage.setItem('SWAPI-Vehicles', JSON.stringify(vehiclesArray));
    }
  }

  getVehiclesData = () => {
    const {pageNumber} = this.state;
    fetchCategoryData('vehicles', pageNumber)
      .then(vehiclesArray => this.setState({vehiclesArray, loading: false}))
      .catch(error => alert(error.message));
  }

  render() {
    const cards = this.state.vehiclesArray.map(card =>
      <VehicleCard
        handleOnClick={this.props.handleOnClick}
        favorites={this.props.favorites}
        key={card.name}
        card={card} />
    );

    return !this.state.loading ?
      (<div className="vehicles-display">
        {cards}
      </div>) :
      (<div className="category-display">
        <img src={loadingGIF} alt="loading" />
      </div>);
  }
}

VehiclesContainer.propTypes = {
  favorites: PropTypes.array.isRequired,
  handleOnClick: PropTypes.func.isRequired
};