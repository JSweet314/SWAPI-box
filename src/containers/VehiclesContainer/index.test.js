import React from 'react';
import {shallow} from 'enzyme';
import VehiclesContainer from './index';

describe('VehicleContainer', () => {
  /* eslint-disable no-undef*/
  const mockHandleOnClick = jest.fn();
  /* eslint-enable no-undef*/
  const mockFavorites = [];

  it('should match a snapshot', () => {
    const wrapper = shallow(
      <VehiclesContainer
        handleOnClick={mockHandleOnClick}
        favorites={mockFavorites} />,
      {disableLifecycleMethods: true}
    );
    expect(wrapper).toMatchSnapshot();
  });
});