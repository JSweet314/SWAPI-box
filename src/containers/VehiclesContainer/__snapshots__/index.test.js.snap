// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleContainer should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VehiclesContainer
    favorites={Array []}
    handleOnClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="loading"
        src="Loading_icon.gif"
/>,
      "className": "category-display",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "loading",
        "src": "Loading_icon.gif",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="loading"
          src="Loading_icon.gif"
/>,
        "className": "category-display",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "loading",
          "src": "Loading_icon.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
